client/admin/components/OrderedFormIterator.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import compose from 'recompose/compose';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import Button from '@material-ui/core/Button';
+import { withStyles } from '@material-ui/core/styles';
+import AddIcon from '@material-ui/icons/AddCircleOutline';
+import { translate } from 'ra-core';
+import { DragDropContext, Droppable } from 'react-beautiful-dnd';
+
+import DraggableFormInput from './DraggableFormInput';
+
+const sanitizeProps = ({ classes, ...props }) => props;
+
+const styles = theme => ({
+  root: {
+    padding: 0,
+    marginBottom: 0,
+    '& > li:last-child': {
+      borderBottom: 'none',
+    },
+  },
+  line: {
+    display: 'flex',
+    listStyleType: 'none',
+    borderBottom: `solid 1px ${theme.palette.divider}`,
+    [theme.breakpoints.down('xs')]: { display: 'block' },
+    '&.fade-enter': {
+      opacity: 0.01,
+      transform: 'translateX(100vw)',
+    },
+    '&.fade-enter-active': {
+      opacity: 1,
+      transform: 'translateX(0)',
+      transition: 'all 500ms ease-in',
+    },
+    '&.fade-exit': {
+      opacity: 1,
+      transform: 'translateX(0)',
+    },
+    '&.fade-exit-active': {
+      opacity: 0.01,
+      transform: 'translateX(100vw)',
+      transition: 'all 500ms ease-in',
+    },
+  },
+  form: { flex: 2 },
+  action: {
+    paddingTop: '0.5em',
+  },
+});
+
+export class OrderedFormIterator extends Component {
+  constructor(props) {
+    super(props);
+    // we need a unique id for each field for a proper enter/exit animation
+    // but redux-form doesn't provide one (cf https://github.com/erikras/redux-form/issues/2735)
+    // so we keep an internal map between the field position and an autoincrement id
+    this.nextId = 0;
```

Comment:
this may become a reason of de-synced order with fields. we should probably use a slugified label as unique field identifier:
```js
import S from 'string';
...
{fields.map((member, index) => {
           const id = S(member.label).slugify().s;
           return (
                    <CSSTransition key={`transition-${id}`} timeout={500} classNames="fade">
                      <DraggableFormInput
                        id={`draggable-${id}`}
                        index={index}
                        member={member}
                        onRemove={this.removeField}
                        {...sanitizeProps(this.props)}
                      />
                    </CSSTransition>
```

Reply:
what label should we use for just now created field?

Reply:
yeah, that's what I mentioned on the call - slugify approach will not really work for all the cases